### ISSUE TITLE
Data -> data and other API inconsistencies

### ISSUE THREAD

**Issue by kwmsmith** [blaze/blaze#1358]
For historical reasons as I understand it, `Data` was originally a class / type in Blaze.  It has since moved to a function.  Nearly everything else in the end-user API is `lowercase_with_underscores`, and there's seemingly no reason besides backwards compatibility for `Data` to be capitalized.

This fix would provide a `data` callable that fills the same role as `Data`.  `Data` would be reimplemented to raise a deprecation warning for 1 minor release cycle, after which it will raise an exception pointing users to `data`, and then it will be removed entirely.

This is related to #1357.

**Comment by kwmsmith** [comment]
Along with this, I vote to deprecate `Symbol` in favor of `symbol`.  Again, once #1357 creates `UnboundSymbol`s, the `Symbol` constructor is no longer needed.

If there are other renamings that we should do, please speak up here.

**Comment by llllllllll** [comment]
I think `Symbol` is already for internal use just the the other expr classes. Users should go through `symbol` for the memoization. One issue is that we import * in init so we expose far to many attributes through our public api.

**Comment by cpcloud** [comment]
`Symbol` and `symbol` are not synonymous. `Symbol` is the type of thing that `symbol` returns, and `symbol` hides some caching behind a function.

**Comment by kwmsmith** [comment]
> Symbol and symbol are not synonymous. Symbol is the type of thing that symbol returns, and symbol hides some caching behind a function.

Right, in #1357 we're proposing renaming the `Symbol` type to `UnboundSymbol`.  With that renaming, I see no reason not to deprecate `Symbol` and point everyone to use `symbol` instead.

**Comment by kwmsmith** [comment]
> I think Symbol is already for internal use just the the other expr classes. Users should go through symbol for the memoization. One issue is that we import * in init so we expose far to many attributes through our public api.

Good point, I'll take a look through our public API and see what we can / should tighten up.

**Comment by g24swint** [comment]
I am super new at using blaze (as in, played with it for only a couple of days), but I like this change as it was puzzling that I didn't end up with a Data after calling Data().

**Comment by kwmsmith** [comment]
Thanks for the input @g24swint.

**Comment by kwmsmith** [comment]
@postelrich  Here's the issue we discussed, should be a quick one.
